### PR TITLE
Refactoring the thread affinity test

### PR DESCRIPTION
**Pull request by krmahadevan**
Refactored the thread affinity test for better
assertions and adding context for failures so 
that failure analysis and debugging can be easy.

### CODE REVIEW

File: core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java [hunk 1]
```diff
@@ -37,29 +40,43 @@ public void testThreadAffinity(Class<?>... classes) {
...
     xmlsuite.setParallel(XmlSuite.ParallelMode.CLASSES);
     xmlsuite.setThreadCount(6);
-    createXmlTest(xmlsuite, "Test 1", classes);
+    createXmlTest(xmlsuite, "Test_1", classes);
     TestNG testng = create(xmlsuite);
     testng.run();
-    assertThat(LogGatheringListener.getLog()).hasSize(2);
+    Map<Class<?>, Set<Long>> map = LogGatheringListener.getLog().get("Test_1");
+    Arrays.stream(classes)
+        .forEach(each -> assertThat(map.get(each))
```
Comment (juherr) [inline core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java:48]: Maybe you can use SoftAssert for that: https://assertj.github.io/doc/#assertj-core-soft-assertions

File: core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java [hunk 2]
```diff
@@ -37,29 +40,43 @@ public void testThreadAffinity(Class<?>... classes) {
...
     xmlsuite.setThreadCount(6);
-    createXmlTest(xmlsuite, "Test_1", PriorityTestSample1.class, PriorityTestSample2.class).setParallel(mode);
-    createXmlTest(xmlsuite, "Test_2", PriorityTestSample1.class,PriorityTestSample2.class).setParallel(mode);
+    Class<?>[] testClasses = new Class<?>[] {PriorityTestSample1.class, PriorityTestSample2.class};
+    createXmlTest(xmlsuite, "Test_1", testClasses).setParallel(mode);
+    createXmlTest(xmlsuite, "Test_2", testClasses).setParallel(mode);
     TestNG testng = create(xmlsuite);
     testng.run();
-    assertThat(LogGatheringListener.getLog()).hasSize(size);
+    for(String test : Arrays.asList("Test_1", "Test_2")) {
```
Comment (juherr) [inline core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java:64]: SoftAssert seems better here too

Comment (krmahadevan) [reply core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java:68]: fixed and pushed amended commit

File: core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java (hunk 1 above)
Comment (krmahadevan) [reply core/src/test/java/test/thread/parallelization/ThreadAffinityTest.java]: fixed and pushed amended commit

**Review by juherr — APPROVED** [review]

File: core/src/test/java/test/thread/parallelization/issue1773/LogGatheringListener.java [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
 
+import java.util.HashSet;
+import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class LogGatheringListener {
 
-    private static Set<Long> log = ConcurrentHashMap.newKeySet();
+    private static Map<String, Map<Class<?>, Set<Long>>> log = new ConcurrentHashMap<>();
```
Comment (vlsi) [inline core/src/test/java/test/thread/parallelization/issue1773/LogGatheringListener.java:10]: make it `final` and use `log.clear()` in reset?

**Pull request merged**